tools/src/main/scala/com/nvidia/spark/rapids/tool/qualification/PluginTypeChecker.scala

```diff
@@ -52,12 +61,131 @@ class PluginTypeChecker {
     writeFormats = writeFormatsTest
   }
 
+  def setOperatorScore(filePath: String): Unit = {
+    val source = Source.fromFile(filePath)
+    supportedOperatorsScore = readOperatorsScore(source)
+  }
+
+  def setSupportedExecs(filePath: String): Unit = {
+    val source = Source.fromFile(filePath)
+    supportedExecs = readSupportedExecs(source)
+  }
+
+  def setSupportedExprs(filePath: String): Unit = {
+    val source = Source.fromFile(filePath)
+    supportedExprs = readSupportedExprs(source)
+  }
+
+  def getOperatorScore: Map[String, Int] = supportedOperatorsScore
+
+  def getSupportedExecs: Map[String, String] = supportedExecs
+
+  def getSupportedExprs: Map[String, String] = supportedExprs
+
+  private def readOperatorsScore: Map[String, Int] = {
+    val source = Source.fromResource(OPERATORS_SCORE_FILE)
+    readOperatorsScore(source)
+  }
+
+  private def readSupportedExecs: Map[String, String] = {
+    val source = Source.fromResource(SUPPORTED_EXECS_FILE)
+    readSupportedExecs(source)
+  }
+
+  private def readSupportedExprs: Map[String, String] = {
+    val source = Source.fromResource(SUPPORTED_EXPRS_FILE)
+    readSupportedExprs(source)
+  }
+
   private def readSupportedTypesForPlugin: (
       Map[String, Map[String, Seq[String]]], ArrayBuffer[String]) = {
     val source = Source.fromResource(DEFAULT_DS_FILE)
     readSupportedTypesForPlugin(source)
   }
 
+  private def readOperatorsScore(source: BufferedSource): Map[String, Int] = {
+    val supportedOperatorsScore = HashMap.empty[String, Int]
+    try {
+      val fileContents = source.getLines().toSeq
+      if (fileContents.size < 2) {
+        throw new IllegalStateException("operatorsScore file appears corrupt," +
+            " must have at least the header and one line")
+      }
+      // first line is header
+      val header = fileContents.head.split(",").map(_.toLowerCase)
+      // the rest of the rows are file with type supported info
+      fileContents.tail.foreach { line =>
+        val cols = line.split(",")
+        if (header.size != cols.size) {
+          throw new IllegalStateException("operatorsScore file appears corrupt," +
+              " header length doesn't match rows length")
+        }
+        val operator = cols(0)
+        val score = cols(1).toInt
+        supportedOperatorsScore.put(operator, score)
+      }
+    } finally {
+      source.close()
+    }
+    supportedOperatorsScore.toMap
+  }
+
+  private def readSupportedExecs(source:BufferedSource): (Map[String, String]) = {
+    val supportedExecs = HashMap.empty[String, String]
```

Comment:
these 3 functions are basically all the same code, perhaps we can combine them. The operators score one we could just do a final map over it to convert the score to Int after the fact.